server/src/main/webapp/WEB-INF/rails/webpack/views/pages/clicky_pipeline_config/tabs/pipeline/materials_widget.tsx

```diff
@@ -159,7 +164,10 @@ export class MaterialsWidget extends MithrilViewComponent<MaterialsAttrs> {
         break;
       case "plugin":
         const pluginAttrs = material.attributes() as PluggableScmMaterialAttributes;
-        const scmMaterial = vnode.attrs.scmMaterials().find((pkg) => pkg.id() === pluginAttrs.ref())!;
+        const scmMaterial = vnode.attrs.scmMaterials().find((pkg) => pkg.id() === pluginAttrs.ref());
+        if(!scmMaterial) {
+          return {};
```

Comment:
Instead of early return, do a break.So you don't have to do a check for undefined at line 109